js/plugins/splittext/plugin.js

```diff
@@ -0,0 +1,350 @@
+   *
+   * @param {object} editor
+   *   CKEditor object.
+   * @param {string} triggerElementName
+   *   Name of trigger element, required for ajax response handling.
+   */
+  Drupal.paragraphs_features.split_text.storeTempData = function (editor, triggerElementName) {
+    var tmpObject = Drupal.paragraphs_features.split_text._tmp;
+
+    var $editorObject = $('#' + editor.name);
+    var selection = editor.getSelection();
+    var ranges = selection.getRanges();
+
+    // Last node that should be selected to cut content.
+    var lastNode = ranges[0].document.getBody().getLast();
+    ranges[0].setEndAfter(lastNode);
+    selection.selectRanges(ranges);
+
+    // Temporal container is used to preserve data over ajax requests.
+    tmpObject.originalEditorSelector = $editorObject.data('drupal-selector');
+
+    // Triggering element is required for proper handling of ajax response.
+    tmpObject.triggeringElementName = triggerElementName;
+
+    // First we "cut" text that will be "pasted" to new added paragraph.
+    tmpObject.newContent = editor.extractSelectedHtml(true);
+    tmpObject.oldContent = editor.getData();
+
+    tmpObject.split_trigger = true;
+  };
+
+  /**
+   * Handler for ajax requests.
+   *
+   * It handles updating of editors are new paragraph is added.
+   *
+   * @param {object} e
+   *   Event object.
+   * @param {object} xhr
+   *   XHR object.
+   * @param {object} settings
+   *   Request settings.
+   */
+  Drupal.paragraphs_features.split_text.onAjaxSplit = function (e, xhr, settings) {
+    // Split text namespace.
+    var stNs = Drupal.paragraphs_features.split_text;
+
+    // Only relevant ajax actions should be handled.
+    if (settings.extraData._triggering_element_name !== stNs._tmp.triggeringElementName || !stNs._tmp.split_trigger) {
+      return;
+    }
+
+    // Set relevant data to original paragraph.
+    var $originalEditor = $('[data-drupal-selector="' + stNs._tmp.originalEditorSelector + '"]');
+    var originalEditor = CKEDITOR.instances[$originalEditor.attr('id')];
+    var $originalRow = $originalEditor.closest('tr');
+    stNs.updateEditor($originalEditor.attr('id'), stNs._tmp.oldContent);
+
+    // Set "cut" data ot new paragraph.
+    var $newRow = $originalRow.nextAll($originalRow.hasClass('odd') ? '.even' : '.odd');
+    var wrapperSelector = stNs.getEditorWrapperSelector(originalEditor);
+    var fieldIndex = $originalEditor.closest('div[data-drupal-selector="' + wrapperSelector + '"]').index();
+    var $newEditor = $($newRow.find('.paragraphs-subform > div')[fieldIndex]).find('textarea');
+    stNs.updateEditor($newEditor.attr('id'), stNs._tmp.newContent);
+
+    // Cleanup states.
+    stNs._tmp.split_trigger = false;
+
+    // Delta field has to be cleaned up for proper working of add button. It
+    // will not make any impact on non modal add mode.
+    $originalRow.closest('table').siblings('.clearfix').find('input.paragraph-type-add-modal-delta').val('');
```

Comment:
I guess we could just remove the clearfix selector.

Reply:
We need strict selector otherwise it could pickup nested paragraph add form.

Reply:
I didn't know that `find()` will work for all `siblings()`, but it does - so this is cleaned up too.